src/webgpu/api/validation/encoding/programmable/pipeline_bind_group_compat.spec.ts

```diff
@@ -1,10 +1,7 @@
 export const description = `
 TODO:
 - test compatibility between bind groups and pipelines
-    - bind groups required by the pipeline layout are required.
-    - bind groups unused by the pipeline layout can be set or not.
-        (Even if e.g. bind groups 0 and 2 are used, but 1 is unused.)
-    - bindGroups[i].layout is "group-equivalent" (value-equal) to pipelineLayout.bgls[i].
+    - the binding resource in bindGroups[i].layout is "group-equivalent" (value-equal) to pipelineLayout.bgls[i].
```

Comment:
Let's keep the spec words: bindGroup.[[layout]] must be group-equivalent with bindGroupLayout.

Reply:
We have done part of group-equivalent in the BGL, such as binding, visibility, and resource type,  now just left the binding resource's group-equivalent validation in TODO which still has many tests needed.